src/main/kotlin/one/oktw/galaxy/proxy/pubsub/Manager.kt

```diff
@@ -1,70 +1,77 @@
 package one.oktw.galaxy.proxy.pubsub
 
-import com.rabbitmq.client.*
+import io.lettuce.core.RedisClient
+import io.lettuce.core.codec.ByteArrayCodec
+import io.lettuce.core.pubsub.RedisPubSubListener
 import one.oktw.galaxy.proxy.Main.Companion.main
 import one.oktw.galaxy.proxy.api.ProxyAPI
 import one.oktw.galaxy.proxy.api.packet.Packet
+import one.oktw.galaxy.proxy.config.CoreSpec
 import one.oktw.galaxy.proxy.event.MessageDeliveryEvent
 import one.oktw.galaxy.proxy.pubsub.data.MessageWrapper
 import java.util.*
-import kotlin.collections.HashMap
+import java.util.concurrent.ConcurrentHashMap
 
-class Manager(private val channel: Channel, private val exchange: String) {
-    companion object {
-        class ConsumerWrapper(private val topic: String, private val manager: Manager) : Consumer {
-            override fun handleRecoverOk(consumerTag: String?) {
+class Manager(prefix: String) {
+    private val client = RedisClient.create(main.config[CoreSpec.redis])
+    private val subscribeConnection = client.connectPubSub(ByteArrayCodec())
+    private val publishConnection = client.connect(ByteArrayCodec())
+    private val queries: ConcurrentHashMap<String, Boolean> = ConcurrentHashMap()
+    private val instanceId: UUID = UUID.randomUUID()
+
+    init {
+        subscribeConnection.addListener(object : RedisPubSubListener<ByteArray, ByteArray> {
```

Comment:
```suggestion
        subscribeConnection.addListener(object : RedisPubSubAdapter<ByteArray, ByteArray>() {
```

And remove unused override